Tallies/TallyClerks/eventClerk_class.f90

```diff
@@ -0,0 +1,470 @@
+        else
+          call self % outputScoresText(id)
+        end if
+      
+        ! Increment the total number of events
+        self % totalEvents = self % totalEvents + self % lastEntry(id)
+        self % lastEntry(id) = 0
+      end if
+      !$omp end critical
+    end if
+
+  end subroutine reportOutColl
+
+  !!
+  !! Display convergence progress on the console
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  subroutine display(self, mem)
+    class(eventClerk), intent(in) :: self
+    type(scoreMemory), intent(in) :: mem
+
+    print *, 'eventClerk does not support display yet'
+
+  end subroutine display
+
+  !!
+  !! Write contents of the clerk to output file.
+  !! In this case, only the map is written to the usual output.
+  !! Meanwhile, the events are written to the event file if they
+  !! haven't already been flushed.
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  subroutine print(self, outFile, mem)
+    class(eventClerk), intent(in)              :: self
+    class(outputFile), intent(inout)           :: outFile
+    type(scoreMemory), intent(in)              :: mem
+    integer(shortInt)                          :: i
+
+    ! Make sure that score arrays have been flushed already.
+    ! If not, write the remainder to output.
+    do i = 1, size(self % lastEntry)
+      if (self % lastEntry(i) > 0) then
+        if (self % binary) then
+          call self % outputScoresBinary(i)
+        else
+          call self % outputScoresText(i)
+        end if
+      end if
+    end do
+
+    ! Write out any map that was used to the usual output file,
+    ! but scores are written out separately.
+
+    ! Begin block
+    call outFile % startBlock(self % getName())
+
+    ! If collision clerk has map print map information
+    if (allocated(self % map)) then
+      call self % map % print(outFile)
+    end if
+    
+    call outFile % endBlock()
+
+  end subroutine print
+
+  !!
+  !! Send events in a thread's bank to a text output file
+  !!
+  subroutine outputScoresText(self, id)
+    class(eventClerk), intent(in) :: self
+    integer(shortInt), intent(in) :: id
+    integer(shortInt)             :: i, u, ios
+
+    associate(ev => self % scores(:,id))
+
+
+      ! Open the file and append the event info
+      open(newunit=u, file=trim(self % outputFile), status='unknown', &
+           position='append', action='write', iostat=ios)
+
+      do i = 1, self % lastEntry(id)
+        ! Consider changing to a binary output in future
+        ! Would require a corresponding python script to translate
+        write(u,'(*(g0,1x))') ev(i) % r, ev(i) % dir, ev(i) % Eincident, &
+                ev(i) % time, ev(i) % w, ev(i) % brood, ev(i) % Edeposit, &
+                ev(i) % MT
+      end do
+      close(u)
+
+    end associate
+
+  end subroutine outputScoresText
+  
+  !!
+  !! Send events in a thread's bank to a binary output file
+  !!
+  subroutine outputScoresBinary(self, id)
```

Comment:
I don't want to be annoying but this could have been done more compactly: the only difference between procedures is access. It could have been one function called outputScores (noting that form = 'unformatted' is default):

if (self % binary):
   accessType = 'stream'
else
   accessType = 'sequential'
end if

open(newunit=u, file=trim(self % outputFile), status='unknown', &
           access=accessType, position='append', &
           action='write', iostat=ios)